drivers/input/input_esp32_touch_sensor.c

```diff
@@ -0,0 +1,390 @@
+		}
+		uint16_t touch_value = touch_hal_read_raw_data(channel_cfg->channel_num);
+
+		touch_hal_set_threshold(channel_cfg->channel_num,
+					touch_value * (100 - channel_cfg->channel_num) / 100);
+	}
+
+#elif defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3)
+	touch_filter_config_t filter_info = {
+		.mode = dev_cfg->filter_mode,
+		.debounce_cnt = dev_cfg->filter_debounce_cnt,
+		.noise_thr = dev_cfg->filter_noise_thr,
+		.jitter_step = dev_cfg->filter_jitter_step,
+		.smh_lvl = dev_cfg->filter_smooth_level,
+	};
+
+	touch_hal_filter_set_config(&filter_info);
+	touch_hal_filter_enable();
+
+	touch_hal_timeout_enable();
+	touch_hal_timeout_set_threshold(SOC_TOUCH_PAD_THRESHOLD_MAX);
+#endif /* defined(CONFIG_SOC_SERIES_ESP32) */
+
+	dev_data->rtc_intr_msk = ESP32_RTC_INTR_MSK;
+	esp32_rtc_isr_install(&esp32_rtc_isr, dev);
+#if defined(CONFIG_SOC_SERIES_ESP32)
+	touch_hal_intr_enable();
+#elif defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3)
+	touch_hal_intr_enable(ESP32_TOUCH_PAD_INTR_MASK);
+	touch_hal_set_fsm_mode(TOUCH_FSM_MODE_TIMER);
+#endif /* defined(CONFIG_SOC_SERIES_ESP32) */
+
+	touch_hal_start_fsm();
+
+	return 0;
+}
+
+#define ESP32_TOUCH_SENSOR_CHANNEL_CFG_INIT(node_id)						\
+	{											\
+		.channel_num  = DT_PROP(node_id, channel_num),					\
+		.channel_sens = DT_PROP(node_id, channel_sens),					\
+		.zephyr_code  = DT_PROP(node_id, zephyr_code),					\
+	}
+
+#define ESP32_TOUCH_SENSOR_INIT(inst)								\
+	static const struct esp32_touch_sensor_channel_config					\
+		esp32_touch_sensor_channel_config_##inst[] = {					\
+			DT_INST_FOREACH_CHILD_STATUS_OKAY_SEP(inst,				\
+				ESP32_TOUCH_SENSOR_CHANNEL_CFG_INIT, (,))			\
+		};										\
+												\
+	static struct esp32_touch_sensor_channel_data esp32_touch_sensor_channel_data_##inst	\
+				[ARRAY_SIZE(esp32_touch_sensor_channel_config_##inst)];		\
+												\
+	static const struct esp32_touch_sensor_config esp32_touch_sensor_config_##inst = {	\
+		.debounce_interval_ms = DT_INST_PROP(inst, debounce_interval_ms),		\
+		.num_channels = ARRAY_SIZE(esp32_touch_sensor_channel_config_##inst),		\
+		.href_microvolt = DT_INST_PROP(inst, href_microvolt),				\
+		.lref_microvolt = DT_INST_PROP(inst, lref_microvolt),				\
+		.href_atten_microvolt = DT_INST_PROP(inst, href_atten_microvolt),		\
+		.filter_mode = DT_INST_PROP(inst, filter_mode),					\
+		.filter_debounce_cnt = DT_INST_PROP(inst, filter_debounce_cnt),			\
+		.filter_noise_thr = DT_INST_PROP(inst, filter_noise_thr),			\
+		.filter_jitter_step = DT_INST_PROP(inst, filter_jitter_step),			\
+		.filter_smooth_level = DT_INST_PROP(inst, filter_smooth_level),			\
+		.channel_cfg = esp32_touch_sensor_channel_config_##inst,			\
+		.channel_data = esp32_touch_sensor_channel_data_##inst,				\
+	};											\
+												\
+	static struct esp32_touch_sensor_data esp32_touch_sensor_data_##inst = {		\
+		.rtc_intr_msk = 0,								\
+	};											\
```

Comment:
no need to initialize this, all static structures are 0 initialized automatically